src/Tribe/Integrations/Restrict_Content_Pro/Service_Provider.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Handles compatibility with Restrict Content Pro plugin.
+ *
+ * @package Tribe\Events\Integrations\Restrict_Content_Pro
+ * @since TBD
+ */
+
+namespace Tribe\Events\Integrations\Restrict_Content_Pro;
+
+use Tribe__Events__Main as TEC;
+
+/**
+ * Integrations with Restrict Content Pro plugin.
+ *
+ * @package Tribe\Events\Integrations
+ *
+ * @since TBD
+ */
+
+class Service_Provider {
+
+	protected static $option_slug = 'tec_events_rcp_hide_on_views';
+
+	/**
+	 * Hooks all the required methods for Restrict Content Pro usage on our code.
+	 *
+	 * @since TBD
+	 *
+	 * @return void  Action hook with no return.
+	 */
+	public function hook() {
+		// Bail when not on V2.
+		if ( ! tribe_events_views_v2_is_enabled() ) {
+			return;
+		}
+
+		// Bail if RCP isn't active.
+		if ( ! function_exists( 'rcp_user_can_access' ) ) {
+			return;
+		}
+
+		// add actions
+		add_action( 'rcp_post_type_restrictions', [ $this, 'add_rcp_post_type_restrictions' ], 10, 2 );
+		add_action( 'rcp_save_post_type_restrictions', [ $this, 'rcp_save_post_type_restrictions' ], 20 );
+		add_action( 'rcp_action_save_post_type_restrictions', [ $this, 'rcp_save_post_type_restrictions' ], 20 );
+
+
+		// add hooks
+		add_filter( 'tribe_template_done', [ $this, 'filter_view_events' ], 20, 3 );
+	}
+
+	public function add_rcp_post_type_restrictions() {
+		$option = tribe_get_option( self::$option_slug, false );
+		?>
+			<p>
+				<label for="<?php echo esc_attr( self::$option_slug ); ?>">
+					<input type="checkbox" name="<?php echo esc_attr( self::$option_slug ); ?>" id="<?php echo esc_attr( self::$option_slug ); ?>" value="1" <?php checked( $option, '1' ); ?>>
+					&nbsp;<?php echo esc_html_x( "Hide restricted events on calendar views.", 'Text label for control to hide restricted events on main calendar views.', 'the-events-calendar' ); ?>
+				</label>
+			</p>
+		<?php
+	}
+
+	public function rcp_save_post_type_restrictions() {
+		if ( empty( $_POST[ self::$option_slug ] ) ) {
```

Comment:
Use `tribe_get_request_var` here to avoid issues with coding standards due to direct access to request super-globals.